### PR TITLE
STYLE: Do not use double underscore defines

### PR DESCRIPTION
**Pull request by hjmjohnson**
From Programming in C++, Rules and Recommendations : The use of two
underscores (`__') in identifiers is reserved for the compiler's
internal use according to the ANSI-C standard.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by seanm** [comment]
Nice, when this is green and merged, I can unsuppress the related clang warning on my bots.

File: Examples/DataRepresentation/Mesh/MeshCellVisitor.cxx [hunk 1]
```diff
@@ -84,8 +84,8 @@ using TetrahedronType = itk::TetrahedronCell<CellType>;
 //  Software Guide : EndLatex
 
 
-#ifndef __CustomTriangleVisitor
-#  define __CustomTriangleVisitor
+#ifndef CUSTOM_TRIANGLE_VISITOR
+#  define CUSTOM_TRIANGLE_VISITOR
```
Comment (dzenanz) [inline Examples/DataRepresentation/Mesh/MeshCellVisitor.cxx:88]: Isn't ITK's style CamelCase?
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkArray.h#L18
Or does that apply only for include guards?

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Comment by seanm** [comment]
So I've removed `-Wno-reserved-id-macro` from one of my bots, let's see tomorrow if it's clean now...

**Comment by seanm** [comment]
Some follow up: https://github.com/InsightSoftwareConsortium/ITK/pull/2969

**Pull request merged**